rcl/src/rcl/event.c

```diff
@@ -0,0 +1,211 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl/event.h"
+
+#include <stdio.h>
+
+#include "rcl/error_handling.h"
+#include "rcl/expand_topic_name.h"
+#include "rcl/remap.h"
+#include "rcutils/logging_macros.h"
+#include "rmw/error_handling.h"
+#include "rmw/validate_full_topic_name.h"
+
+#include "./common.h"
+#include "./types_impl.h"
+
+
+rcl_event_t
+rcl_get_zero_initialized_event()
+{
+  static rcl_event_t null_event = {0};
+  return null_event;
+}
+
+rcl_ret_t
+rcl_publisher_event_init(
+  rcl_event_t * event,
+  const rcl_publisher_t * publisher,
+  const rcl_publisher_options_t * options,
```

Comment:
Seems like overkill passing in all the options when you just need the allocator and the allocator isn't considered optional for this function. I think we should break out the allocator into a separate parameter that makes it clear it's required. 

Same applies to all the below.

Reply:
I will pass in a `const rcl_allocator_t *` instead of a `rcl_*_options_t *`.